src/github.com/getlantern/flashlight/util/http.go

```diff
@@ -36,10 +36,17 @@ func success(resp *http.Response) bool {
 	return resp.StatusCode > 199 && resp.StatusCode < 400
 }
 
-// NewChainedAndFronted creates a new struct for accessing resources using chained
-// and direct fronted servers in parallel.
-func NewChainedAndFronted(proxyAddrFN eventual.Getter) *chainedAndFronted {
-	cf := &chainedAndFronted{proxyAddrFN: proxyAddrFN}
+// NewChainedAndFronted creates a new struct for accessing resources using
+// chained and direct fronted servers. If parallel is true, the request will be
+// sent to both servers. If false, we'll first try the chained option and only
+// fall back to direct fronting if chained fails. Once a chained requets
+// succeeds, we only used chained going forward unless it fails, at which point
+// we go back to the original regime.
+func NewChainedAndFronted(proxyAddrFN eventual.Getter, parallel bool) *chainedAndFronted {
```

Comment:
I'm not particularly fond of this API. I'd use a constant with a name that clearly states the strategy
http.ChainedAndFrontedParallelStrategy
http.ChainedAndFrontedSequentialStrategy

There might be other strategies in the future, but more importantly, I think we could gain in clarity, instead of using a large comment (which is only found in the definition of the function, not in every usage).